wolfcrypt/src/asn.c

```diff
@@ -8730,97 +8738,41 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
                 if (decrementMaxPathLen && cert->ca->maxPathLen > 0) {
                     WOLFSSL_MSG("\tmaxPathLen status: reduce by 1");
                     cert->maxPathLen = cert->ca->maxPathLen - 1;
+                    if (verify != NO_VERIFY && type != CA_TYPE &&
+                                                    type != TRUSTED_PEER_TYPE) {
+                        WOLFSSL_MSG("\tmaxPathLen status: OK");
+                    }
                 } else if (decrementMaxPathLen && cert->ca->maxPathLen <= 0) {
-                    /* Will be handled as ERROR in "verify check" below */
                     cert->maxPathLen = 0;
+                    if (verify != NO_VERIFY && type != CA_TYPE &&
+                                                    type != TRUSTED_PEER_TYPE) {
+                        WOLFSSL_MSG("\tNon-entity cert, maxPathLen is 0");
+                        WOLFSSL_MSG("\tmaxPathLen status: ERROR");
+                        return ASN_PATHLEN_INV_E;
+                    }
                 }
             } else if (cert->ca && cert->isCA) {
                 /* case where cert->pathLength extension is not set */
                 if (cert->ca->maxPathLen > 0) {
                     cert->maxPathLen = cert->ca->maxPathLen - 1;
                 } else {
                     cert->maxPathLen = 0;
-                }
-            }
-        }
-
-        if (verify != NO_VERIFY && type != CA_TYPE &&
```

Comment:
There is a huge section of code here removed including some useful comments, OCSP only checks, "About to verify certificate signature" message, etc... Was this purposeful and if so how can explain why and how we can ensure this doesn't break existing use cases? Specifically logic for `WOLFSSL_NO_TRUSTED_CERTS_VERIFY` and OCSP call to `GetCABySubjectAndPubKey`.

Reply:
The code has been merged.
It was doing the lookup for the CA twice and then duplicating the path length checking.

Reply:
Looking at GetCABySubjectAndPubKey I can't see an issue.
It will be called in the same conditions.

Reply:
@SparkiDev : Okay I see now. The code removed below was duplicate checks added with the path length PR. The `WOLFSSL_NO_TRUSTED_CERTS_VERIFY` and `GetCABySubjectAndPubKey` logic still exists above at line 8674.